### ISSUE TITLE
treat echo in CLI as a different taint-sink

### ISSUE THREAD

**Issue by staabm** [vimeo/psalm#5132]
Should code like

```php
   If (PHP_SAPI == 'cli') {
     echo $string;
   }
```

be treated as a shell-taint sink? It seems it currently treated as a html-sink?

**Comment by weirdan** [comment]
`shell` sink is for the shell commands as far as I'm aware. Does Psalm have something like `terminal` sync (as you can do some nasty things with ANSI escapes)?

**Comment by muglug** [comment]
@staabm at Vimeo we just suppress the `TaintedHtml` error in folders that contain CLI scripts.

**Comment by gharlan** [comment]
In our case the code is part of the error-handler which is used in web and cli.
My current proposal was to use `@psalm-taint-escape html`, which is a bit weird in that case: https://github.com/redaxo/redaxo/pull/4381/files

**Comment by muglug** [comment]
ah ok – that's the sort of place where a more custom system like CodeQL might allow you to craft a rule that could exclude false positives and also false negatives (but no such thing is one the horizon for Psalm)
